### DISCUSSION TITLE
Optional case sensitivity?

### DISCUSSION THREAD

**Discussion by PTKu** [simatic-ax/.discussions#17] [Ideas]
I know, I know... there is no such thing in IEC61131-3 😁
But would it be possible to introduce it as an option? 
There are many instances where one would call an argument and a member with the same name but a different casing, which is a sensible thing to do in many cases.😉

**Comment by sjuergen** [comment]
Hi @PTKu thanks for sharing always your new ideas. But do you think, it makes sense to implement such things in a programming language which is by definition not case sensitive? @theBadT what do you think?

**Reply by PTKu** [reply]
Hey @sjuergen, I think it would be helpful if someone wants to follow naming conventions where casing matters.

In the following 

~~~
CLASS FooClass
        VAR PUBLIC 
            Foo : BOOL;
        END_VAR

        METHOD FooMethod : Int
            VAR_INPUT
                foo : BOOL;
            END_VAR
            ;
        END_METHOD
    END_CLASS
~~~

Public `Foo` and input argument `foo` is the same variable, and indeed:

![image](https://github.com/simatic-ax/.discussions/assets/61538034/6ba76842-f2db-4c40-b698-5e73757fd72c)

**Reply by sjuergen** [reply]
@PTKu I know what you mean. I had also in some cases the wish like yours. But I extended your example a little bit. And I come to the conclusion, that it is very hard to read and probably a high risk for errors.

```
CLASS FooClass
        VAR PUBLIC 
            Foo : BOOL;
        END_VAR

        METHOD FooMethod : Int
            VAR_INPUT
                foo : BOOL;
            END_VAR
            VAR_OUTPUT
               foO : BOOL;
            END_VAR
            IF (foo = Foo) THEN
               foO = (Foo OR foo);
            END_IF;
        END_METHOD
    END_CLASS
```

**Reply by sjuergen** [reply]
But I'm also not a fan of prefixed like:
```
I_Foo
O_Foo
iFoo
oFoo
```

**Reply by PTKu** [reply]
The idea is that if one wants to apply naming convention where the arguments of a method are using camel casing, there is good chance for a conflict with some public member.
